internetarchive/cli/ia.py

```diff
@@ -117,16 +114,16 @@ def main():
 
     # Validate args.
     s = Schema({
-        six.text_type: bool,
-        '--config-file': Or(None, *six.string_types),
-        '--host': Or(None, *six.string_types),
+        str: bool,
+        '--config-file': Or(None, str),
+        '--host': Or(None, str),
         '<args>': list,
         '<command>': Or(str, lambda _: 'help'),
     })
     try:
         args = s.validate(args)
     except SchemaError as exc:
-        print('{0}\n{1}'.format(str(exc), printable_usage(__doc__)), file=sys.stderr)
+        print(f'{exc!s}\n{printable_usage(__doc__)}', file=sys.stderr)
```

Comment:
The cast to string here is very terse, do we really want this instead of `str(exc)`?

If we switch to the `f''` syntax we might as well embrace it fully, but this is still something that we should consider first.

Reply:
Definitely in favour of embracing it. Another option would be to get rid of these string casts because variables always get cast to string inside an f-string anyway when you don't specify a conversion operator.

Reply:
Good point, @JustAnotherArchivist, let's get rid of these string casts when they're not needed.

When it is needed, I agree that we should embrace `f''` syntax fully.

Reply:
Minor correction: the default conversion happens using the variable's `__format__` method. Because neither `schema.SchemaError` nor anything in its parent class tree defines such a method, the top-level `object.__format__` is used, which [simply returns `str(self)`](https://github.com/python/cpython/blob/a0e55a571cf01885fd5826266c37abaee307c309/Objects/typeobject.c#L5532). The same thing holds for the other cases where the `!s` conversion is currently used. Will remove shortly.